### PR TITLE
VFL spark DataFrame API for scala

### PR DESCRIPTION
**Pull request by Litchilitchy**

### CODE REVIEW

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/vfl/FLModel.scala [hunk 1]
```diff
@@ -19,21 +19,25 @@ package com.intel.analytics.bigdl.ppml.vfl
...
+import com.intel.analytics.bigdl.ppml.utils.DataFrameUtils
 import com.intel.analytics.bigdl.ppml.vfl.nn.VflNNEstimator
+import org.apache.spark.sql.DataFrame
 
 class FLModel() {
   val estimator: VflNNEstimator = null
-  def fit(trainData: DataSet[MiniBatch[Float]],
-          valData: DataSet[MiniBatch[Float]],
+  def fit(trainData: DataFrame,
+          valData: DataFrame,
```
Comment (jason-dai) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/vfl/FLModel.scala:29]: need to specify feature columns and (optionally) label columns, similar to Orca estimator

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/vfl/FLModel.scala [hunk 2]
```diff
@@ -19,21 +19,25 @@ package com.intel.analytics.bigdl.ppml.vfl
...
+    val _valData = DataFrameUtils.dataFrameToSample(valData)
+    estimator.train(epoch, _trainData.toLocal(), _valData.toLocal())
   }
   def evaluate() = {
     estimator.getEvaluateResults().foreach{r =>
       println(r._1 + ":" + r._2.mkString(","))
     }
   }
-  def predict(data: DataSet[MiniBatch[Float]]) = {
+  def predict(data: DataFrame) = {
```
Comment (jason-dai) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/vfl/FLModel.scala:40]: need to specify feature columns, similar to Orca estimator

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/utils/DataFrameUtils.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+import com.intel.analytics.bigdl.dllib.feature.dataset.{DataSet, MiniBatch, Sample, SampleToMiniBatch}
+import com.intel.analytics.bigdl.dllib.tensor.Tensor
+import com.intel.analytics.bigdl.ppml.vfl.VflContext
+import com.intel.analytics.bigdl
+import org.apache.spark.sql.types.FloatType
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+object DataFrameUtils {
+  def dataFrameToSample(df: DataFrame, isTrain: Boolean = true, batchSize: Int = 4): bigdl.DataSet[MiniBatch[Float]] = {
```
Comment (jason-dai) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/utils/DataFrameUtils.scala:28]: need to do something similar to https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/orca/src/bigdl/orca/learn/pytorch/pytorch_spark_estimator.py#L68?

**Comment by Litchilitchy** [comment]
Changes are in #3685

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/example/logisticregression/VflLogisticRegression.scala [hunk 4]
```diff
@@ -100,7 +70,6 @@ object VflLogisticRegression {
       opt[String]('r', "rowKeyName")
         .text("row key name of data")
         .action((x, params) => params.copy(rowKeyName = x))
-        .required()
       opt[String]('l', "learningRate")
         .text("learning rate of training")
         .action((x, params) => params.copy(learningRate = x.toFloat))
```
Comment (jason-dai) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/example/logisticregression/VflLogisticRegression.scala:75]: change below to `FlContext.initFlContext`

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/vfl/FLModel.scala (hunk 1 above)
Comment (jason-dai) [reply scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/vfl/FLModel.scala:29]: The API should be similar to https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/orca/src/bigdl/orca/learn/pytorch/pytorch_spark_estimator.py#L104

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/example/logisticregression/VflLogisticRegression.scala (hunk 4 above)
Comment (Litchilitchy) [reply scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/example/logisticregression/VflLogisticRegression.scala:75]: This is in #3687

**Pull request merged**